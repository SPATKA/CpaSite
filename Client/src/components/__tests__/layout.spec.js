import React from "react"
import renderer from "react-test-renderer"
import Layout from "../layout"
import Header from "../header"
import Footer from "../footer"
import TestRenderer from "react-test-renderer"
import * as Gatsby from "gatsby"

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`)
const mockUseStaticQuery = {
  site: {
    siteMetadata: {
      title: `KNaik Associates`,
      menuLinks: [],
    },
  }
}

describe("Layout", () => {

  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }))
    });
  });

  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery)
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it("renders correctly", () => {
    const tree = renderer.create(<Layout />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("should renders `Header` & `Footer' components", () => {
    useStaticQuery.mockReturnValueOnce({
      site: {
        siteMetadata: {
          title: `KNaik Associates`,
          menuLinks: [],
        },
      },
    })

    const testRenderer = TestRenderer.create(<Layout />)
    const testInstance = testRenderer.root
    expect(testInstance.findByType(Header)).toBeTruthy()
    expect(testInstance.findByType(Footer)).toBeTruthy()
  })
})
