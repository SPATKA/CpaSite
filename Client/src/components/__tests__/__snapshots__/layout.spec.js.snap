// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <div
    className="d-flex justify-content-between d-lg-none py-2"
  >
    <a
      className="flex-grow-1 text-center icon-color"
      href="https://knaikassociates.securefilepro.com/portal/#/login"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-lock fa-sm "
        data-icon="lock"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M144 144v48H304V144c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192V144C80 64.5 144.5 0 224 0s144 64.5 144 144v48h16c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V256c0-35.3 28.7-64 64-64H80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <a
      className="flex-grow-1 text-center icon-color"
      href="mailto:Kartiki@knaikassociates.com"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-envelope fa-sm "
        data-icon="envelope"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <a
      className="flex-grow-1 text-center icon-color disabledLink"
      href="tel:(973) 462-1499"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-phone fa-sm "
        data-icon="phone"
        data-prefix="fas"
        data-testid="fa-icon"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M164.9 24.6c-7.7-18.6-28-28.5-47.4-23.2l-88 24C12.1 30.2 0 46 0 64C0 311.4 200.6 512 448 512c18 0 33.8-12.1 38.6-29.5l24-88c5.3-19.4-4.6-39.7-23.2-47.4l-96-40c-16.3-6.8-35.2-2.1-46.3 11.6L304.7 368C234.3 334.7 177.3 277.7 144 207.3L193.3 167c13.7-11.2 18.4-30 11.6-46.3l-40-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <span
      className="flex-grow-1 text-center icon-color"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-sm "
        data-icon="bars"
        data-prefix="fas"
        data-testid="fa-icon-menu"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>,
  <div
    className="offcanvas offcanvas-lg offcanvas-end"
  >
    <div
      className="d-lg-none offcanvas-header"
    >
      <div
        className="offcanvas-title h5"
      >
        Menu
      </div>
      <button
        aria-label="Close"
        className="btn-close"
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="p-0 d-lg-none offcanvas-body"
    >
      <div
        className="sidebar"
      />
    </div>
  </div>,
  <div
    className="d-none d-lg-flex justify-content-lg-between py-2 container"
  >
    <div>
      <a
        href="https://knaikassociates.securefilepro.com/portal/#/login"
      >
        <button
          className="btn btn-default btn-sm"
        >
          Login
        </button>
      </a>
    </div>
    <h3
      className="strong lead"
    >
      Call Us: 
      <a
        className="undefined disabledLink"
        href="tel:(973) 462-1499"
      >
        (973) 462-1499
      </a>
    </h3>
  </div>,
  <div
    className="d-flex justify-content-center justify-content-lg-between py-2 container"
  >
    <div
      className="d-none d-lg-flex my-auto"
    >
      <ul
        className="linkContainer"
      />
    </div>
  </div>,
  <main />,
  <footer>
    <div
      className="row text-center mx-auto f-card-container container"
    >
      <div
        className="col-lg-4 col-md-6 my-3 mx-auto text-center"
        data-sal="zoom-in"
        data-sal-duration="2000"
        data-sal-easing="ease"
      >
        <div
          className="icon-color"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-comment fa-5x "
            data-icon="comment"
            data-prefix="far"
            data-testid="fa-icon"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32C114.6 32 .0272 125.1 .0272 240c0 47.63 19.91 91.25 52.91 126.2c-14.88 39.5-45.87 72.88-46.37 73.25c-6.625 7-8.375 17.25-4.625 26C5.818 474.2 14.38 480 24 480c61.5 0 109.1-25.75 139.1-46.25C191.1 442.8 223.3 448 256 448c141.4 0 255.1-93.13 255.1-208S397.4 32 256 32zM256.1 400c-26.75 0-53.12-4.125-78.38-12.12l-22.75-7.125l-19.5 13.75c-14.25 10.12-33.88 21.38-57.5 29c7.375-12.12 14.37-25.75 19.88-40.25l10.62-28l-20.62-21.87C69.82 314.1 48.07 282.2 48.07 240c0-88.25 93.25-160 208-160s208 71.75 208 160S370.8 400 256.1 400z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <h3>
          Ask a Question
        </h3>
        <p>
          Find comfort in knowing an Expert in accounting is only an email or phone-call away.
        </p>
        <button
          className="btn btn-primary btn-lg"
        >
          <a
            className="undefined disabledLink"
            href="tel:(973) 462-1499"
          >
            (973) 462-1499
          </a>
        </button>
      </div>
      <div
        className="col-lg-4 col-md-6 my-3 mx-auto text-center"
        data-sal="zoom-in"
        data-sal-duration="2000"
        data-sal-easing="ease"
      >
        <div
          className="icon-color"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-calendar fa-5x "
            data-icon="calendar"
            data-prefix="far"
            data-testid="fa-icon"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M152 64H296V24C296 10.75 306.7 0 320 0C333.3 0 344 10.75 344 24V64H384C419.3 64 448 92.65 448 128V448C448 483.3 419.3 512 384 512H64C28.65 512 0 483.3 0 448V128C0 92.65 28.65 64 64 64H104V24C104 10.75 114.7 0 128 0C141.3 0 152 10.75 152 24V64zM48 448C48 456.8 55.16 464 64 464H384C392.8 464 400 456.8 400 448V192H48V448z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <h3>
          We Are Here to Help
        </h3>
        <p>
          We will happily offer you a free consultation to determine how we can best serve you.
        </p>
        <button
          className="btn btn-primary btn-lg"
        >
          <a
            href="/contact"
          >
            Contact Us
          </a>
        </button>
      </div>
      <div
        className="col-lg-4 col-md-6 my-3 mx-auto text-center"
        data-sal="zoom-in"
        data-sal-duration="2000"
        data-sal-easing="ease"
      >
        <div
          className="icon-color"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-copy fa-5x "
            data-icon="copy"
            data-prefix="far"
            data-testid="fa-icon"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M502.6 70.63l-61.25-61.25C435.4 3.371 427.2 0 418.7 0H255.1c-35.35 0-64 28.66-64 64l.0195 256C192 355.4 220.7 384 256 384h192c35.2 0 64-28.8 64-64V93.25C512 84.77 508.6 76.63 502.6 70.63zM464 320c0 8.836-7.164 16-16 16H255.1c-8.838 0-16-7.164-16-16L239.1 64.13c0-8.836 7.164-16 16-16h128L384 96c0 17.67 14.33 32 32 32h47.1V320zM272 448c0 8.836-7.164 16-16 16H63.1c-8.838 0-16-7.164-16-16L47.98 192.1c0-8.836 7.164-16 16-16H160V128H63.99c-35.35 0-64 28.65-64 64l.0098 256C.002 483.3 28.66 512 64 512h192c35.2 0 64-28.8 64-64v-32h-47.1L272 448z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <h3>
          Send Us a File
        </h3>
        <p>
          Use our convenient SecureSend page to securely deliver a file directly to a member of our firm.
        </p>
        <button
          className="btn btn-primary btn-lg"
        >
          <a
            href="https://knaikassociates.securefilepro.com/portal/#/login"
          >
            Secure Send
          </a>
        </button>
      </div>
    </div>
    <div
      className="footer-nav"
    >
      <div
        className="container"
      >
        <div
          className="linkConatainer"
        >
          <ul
            className="navLinks"
          >
            <li>
              <h3>
                Contact
              </h3>
            </li>
            <li>
              K Naik Associates
            </li>
            <li>
              <a
                className="undefined disabledLink"
                href="tel:(973) 462-1499"
              >
                (973) 462-1499
              </a>
            </li>
            <li>
              <a
                href="mailto:Kartiki@knaikassociates.com"
              >
                Kartiki@knaikassociates.com
              </a>
            </li>
          </ul>
        </div>
        <hr />
        <div
          className="linkConatainer"
        >
          <ul
            className="linkConatainerB"
          >
            <li
              className="ps-2"
            >
              <span>
                © 
                K Naik Associates
                 2022
              </span>
            </li>
            <li
              className="ps-2"
            >
              <a
                href="?"
              >
                Site Map
              </a>
            </li>
            <li
              className="ps-2"
            >
              <a
                href="?"
              >
                Privacy Policy
              </a>
            </li>
            <li
              className="ps-2"
            >
              <a
                href="?"
              >
                Disclaimer
              </a>
            </li>
          </ul>
          <div
            className="text-center"
          >
            <a
              className="m-1"
              href="https://www.facebook.com"
              title="Facebook"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-facebook "
                data-icon="facebook"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
            <a
              className="m-1"
              href="https://www.twitter.com"
              title="Twitter"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-twitter "
                data-icon="twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
            <a
              className="m-1"
              href="https://www.linkedin.com"
              title="LinkedIn"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-linkedin "
                data-icon="linkedin"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>,
]
`;
